src/graph/on_disk_graph.cpp

```diff
@@ -0,0 +1,71 @@
+#include "graph/on_disk_graph.h"
+
+#include "storage/storage_manager.h"
+
+using namespace kuzu::catalog;
+using namespace kuzu::storage;
+using namespace kuzu::main;
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace graph {
+
+NbrScanState::NbrScanState(storage::MemoryManager* mm) {
+    srcNodeIDVectorState = DataChunkState::getSingleValueDataChunkState();
+    dstNodeIDVectorState = std::make_shared<common::DataChunkState>();
+    srcNodeIDVector = std::make_unique<ValueVector>(*LogicalType::INTERNAL_ID(), mm);
+    srcNodeIDVector->state = srcNodeIDVectorState;
+    dstNodeIDVector = std::make_unique<ValueVector>(*LogicalType::INTERNAL_ID(), mm);
+    dstNodeIDVector->state = dstNodeIDVectorState;
+    fwdReadState = std::make_unique<RelTableReadState>(columnIDs, direction);
+    fwdReadState->nodeIDVector = srcNodeIDVector.get();
+    fwdReadState->outputVectors.push_back(dstNodeIDVector.get());
+}
+
+OnDiskGraph::OnDiskGraph(main::ClientContext* context, const std::string& nodeName,
+    const std::string& relName)
+    : context{context} {
+    auto catalog = context->getCatalog();
+    auto storage = context->getStorageManager();
+    auto tx = context->getTx();
```

Comment:
Ouch, you guys are hurting me by abbreviating transaction as tx. Can we do txn? Both in the getTx() function and in tx variables.

Reply:
@ray6080 will change this.